json/json_path.mbt

```diff
@@ -31,17 +31,24 @@ pub fn add_key(self : JsonPath, key : String) -> JsonPath {
 
 ///|
 pub impl Show for JsonPath with output(self, logger) {
-  match self {
-    Root => logger.write_string("$")
-    Key(p, key~) =>
-      logger..write_object(p)..write_string(".")..write_string(key)
-    Index(p, index~) =>
-      logger
-      ..write_object(p)
-      ..write_string("[")
-      ..write_object(index)
-      ..write_string("]")
+  // Helper function to escape JSON Pointer tokens
+  fn escape_token(token : String) -> String {
+    // Replace ~ with ~0, then / with ~1
+    // Order matters: ~ must be escaped first
+    token.replace(old="~", new="~0").replace(old="/", new="~1")
   }
+
+  // Recursively build the JSON Pointer string
+  fn build_path(path : JsonPath) -> String {
```

Comment:
note this implementation may be inefficient.
 @peter-jerry-ye is this what you expected? display JSON Path using JSON pointer notation